armi/reactor/tests/test_blocks.py

```diff
@@ -1242,6 +1242,11 @@ def test_getNumPins(self):
         emptyBlock = blocks.HexBlock("empty")
         self.assertEqual(emptyBlock.getNumPins(), 0)
 
+        comp = basicShapes.Hexagon("hexagon", "HT9", 1, 1, 1)
```

Comment:
I love it!

You've added so much logic to `getNumPins()` I wonder if we can test it a bit more.

For instance, can we have one test where it returns a NON-zero pin count, because it has a circular shape somewhere in it?  Perhaps we could do one with a `HoledRectangle`, and prove that the hole in the rectangle?

https://github.com/terrapower/armi/blob/20ca6877c2e2221bc3e3521cf9e827cf2d4a4376/armi/reactor/components/complexShapes.py#L94

Reply:
I added some more tests. Let me know if you think there are other corner cases we should cover!